### DISCUSSION TITLE
Blog on using Matter.js to control a light

### DISCUSSION THREAD

**Discussion by potto216** [matter-js/matter.js#1156] [Show and tell]
Hi All, I wrote a blog on using Matter.js to control a light https://medium.com/@potto_94870/turning-on-a-light-with-javascript-dedd8a8a4ab8 . The article should be accessible by anyone without a Medium account. I appreciate suggestions and corrections.

**Comment by Apollon77** [comment]
Hi Paul,

oh thats great news!

And yes some comments and maybe possible enhancements:
* A "Fabric" is not really something network defined or bound. A Fabric in Matter is a "Cryptographically secured sub environment where the included nodes share the same cryptographical roots and so can interoperate with each other, but are so cryptographically separated from other fabrics". Also a fabric is just  a detail of the whole matter spec and so I think it would be better to talk about Matter iot standard and refer to the download instead of fabrics. Each controller/ecosystem will setup it's own fabric
* Ipv6 internet access is not the topic ... IPv6 local access and "having it enabled in the local network" is all about it. Matter is a local protocol and so not suitable for usage over internet anyway, but IPv6 is a MUST. The info on "ipv6 internet access" might let people miss the local aspect even if told in next section
* if you like to add info on how use matter.js as package then I would propose to install the matter-node.js-examples npm package 
* Please check the CLI you use "npm run matter-device -- -type socket --on="/home/user/relay.sh CH1 OFF" --off="/home/user/relay.sh CH1 ON" -loglevel debug" ... it might need to be changed for type and loglevel to match new formats for 0.10 which was just released
* "Commissioning Raspberry Pi onto the Matter Network" is not 100% accurate ... because if fact you commission the "device you just started using the CLI" because in the case of matter.js you could strat multiple on different ports and each one would be an own node ...
* Especially for Google (when using Goole via ios) you need to do steps to whitelist the test device for google usage. See matter.js readme, see https://github.com/project-chip/matter.js/blob/main/docs/ECOSYSTEMS.md#google-home-ecosystem

I hope my comments help. If of I would also in some days share your article via my LinkedIn?

**Reply by potto216** [reply]
@Apollon77 thanks for the detailed feedback. Your explanation of the Fabric is helpful when reading Section 7.5 of the Core spec. I'm going to work on the edits over the next few days in particular correcting the commissioning of a Raspberry Pi. So, from a commissioning standpoint do you think of the software running one or more nodes as the device? 

The term "device commissioning" is confusing to me because the Matter core spec defines a device as "A piece of equipment containing one or more Nodes", but I see where you can have different nodes of the same device on different Fabrics which would require their own commissioning.

Once I get the edits done and rerun the scripts on version 0.10 it would be awesome to share the article over LinkedIn.

**Reply by Apollon77** [reply]
Re Nodes: Normally, when considering end user devices, you have one hardware offering one node. but that's mainly limited by the "what makes sense for end users" and also capabilities of systems like ESP32 based things and such. In theory you can operate one node for each "port number" you can choose - with the speciality that there is only 5540 officially assigned as Matter port number and e.g. Amazon hard limits on this. But with all others you can simply run multiple matter.js processes using different ports and have fun with one node per port :-)
matter.js allows you to handle multiple ports also within one nodejs process - or you start multiple but have some MDNS overhead with this because each process will broadcast and scan all the MDNS stuff.

Also Fabrics and nodes are separated .. so you can pair one node to multiple fabrics!

**Reply by potto216** [reply]
@Apollon77 I made the edits concerning 
- correcting the definition of fabric and commissioning along with better defining nodes and devices.
- Removing the IPv6 Internet access from the Wi-Fi tip and clarifying the Matter is a local protocol.

Let me know of other issues and thanks for the review.
https://medium.com/@potto_94870/turning-on-a-light-with-javascript-dedd8a8a4ab8

**Reply by Apollon77** [reply]
If you checked the cli command used (parameters with one dash vs two dashes!) then all is fine

**Comment by potto216** [comment]
Yep I ran it with the main pull yesterday.

**Comment by Apollon77** [comment]
Alright, I might post the article monday or such on LinkedIn...

**Reply by Apollon77** [reply]
Then happy re-sharing :-)
https://www.linkedin.com/posts/ifischer_turning-on-a-light-with-javascript-activity-7239892641629286400-iE3E
